src/SystemCommands-MethodCommands/SycProfileMethodCommand.class.st

```diff
@@ -0,0 +1,100 @@
+"
+Provides a method selection menu item to profile its execution. If the method is a regular method, it will be executed as is, without parameter provision. If the method is a test method, it will be executed within its tests class receiver through the `run:` message.
+
+The execution should open a Profiler Presenter window with the results of the profiling.
+
+Multiple method selections will open multiple Profiler Presenters.
+
+If there are multiple classes selected, the command will be aborted.
+"
+Class {
+	#name : 'SycProfileMethodCommand',
+	#superclass : 'SycMethodCommand',
+	#category : 'SystemCommands-MethodCommands',
+	#package : 'SystemCommands-MethodCommands'
+}
+
+{ #category : 'accessing' }
+SycProfileMethodCommand >> defaultMenuIconName [
+
+	^ #smallProfile
+]
+
+{ #category : 'accessing' }
+SycProfileMethodCommand >> defaultMenuItemName [
+
+	^ 'Profile method'
+]
+
+{ #category : 'execution' }
+SycProfileMethodCommand >> execute [
+
+	methods do: [ : selector | self profileMethod: selector ].
+
+]
+
+{ #category : 'execution' }
+SycProfileMethodCommand >> newProfilerPresenter [
+
+	^ self profilerPresenterClass new
+]
+
+{ #category : 'execution' }
+SycProfileMethodCommand >> profileMethod: aCompiledMethod [ 
+	
+	aCompiledMethod isTestMethod
+		ifTrue: [ self profileTestMethod: aCompiledMethod ]
+		ifFalse: [ self profileRegularMethod: aCompiledMethod  ]
+]
+
+{ #category : 'execution' }
+SycProfileMethodCommand >> profileRegularMethod: aCompiledMethod [ 
```

Comment:
This will not work all the times. The receiver can be an object that requires complex logic to be initialized

Reply:
Of course it won't work all the time. It depends on the method implementation, so it requires some minimum knowledge by the user. Because you can write a method

```methodToProfile

   ^1000000000 factorial
 ```

But if you want to make a complete and pedantic fool-proof implementation nothing is stopping you 😛